android/titanium/src/java/org/appcelerator/titanium/io/TiFileFactory.java

```diff
@@ -122,6 +154,34 @@ private static String formPath(String path, String parts[])
 		return path;
 	}
 
+	/**
+	 * Removes all characters matching the given "trimCharacter" from the front of given "sourceString"
+	 * and returns the result.
+	 * @param sourceString The string to have characters trimmed from. Can be empty or null.
+	 * @param trimCharacter The character to be trimmed from given "sourceString".
+	 * @return
+	 * Returns a new string if characters were trimmed from the front.
+	 * Returns the same string reference if no characters were trimmed.
+	 * Returns null if given a null string.
+	 */
+	private static String trimFront(String sourceString, char trimCharacter)
```

Comment:
I was able to simplify this a little:
```Java
private static String trimFront(String sourceString, char trimCharacter)
{
	if (sourceString != null) {
		for (int i = 0; i < sourceString.length(); i++) {
			if (sourceString.charAt(i) != trimCharacter) {
				sourceString = sourceString.substring(i);
				break;
			}
		}
	}
	return sourceString;
}
```

Reply:
This doesn't handle the case where all characters in the string are trimmable. That's why I do the substring handling outside of the loop.

Reply:
I think that a `for` is a little easier to read than a `while` in this case:
```Java
private static String trimFront(String sourceString, char trimCharacter)
{
  if (sourceString != null) {
    int i = 0;
    for (; i < sourceString.length(); i++) {
      if (sourceString.charAt(i) != trimCharacter) {
        sourceString = sourceString.substring(i);
        break;
      }
    }
    if (i == sourceString.length()) {
      sourceString = "";
    }
  }
  return sourceString;
}
```

But they both work, so I'll let you decide.

Reply:
I originally wrote it with a `for` like this...
```java
for (; (index < sourceString.length()) && (sourceString.charAt(index) == trimCharacter); index++);
```

But our lint step didn't like it. _(shrugs)_